src/blocks/scratch3_data.js

```diff
@@ -0,0 +1,180 @@
+var Cast = require('../util/cast');
+
+function Scratch3DataBlocks(runtime) {
+    /**
+     * The runtime instantiating this block package.
+     * @type {Runtime}
+     */
+    this.runtime = runtime;
+}
+
+/**
+ * Retrieve the block primitives implemented by this package.
+ * @return {Object.<string, Function>} Mapping of opcode to Function.
+ */
+Scratch3DataBlocks.prototype.getPrimitives = function () {
+    return {
+        'data_variable': this.getVariable,
+        'data_setvariableto': this.setVariableTo,
+        'data_changevariableby': this.changeVariableBy,
+        'data_list': this.getList,
+        'data_addtolist': this.addToList,
+        'data_deleteoflist': this.deleteOfList,
+        'data_insertatlist': this.insertAtList,
+        'data_replaceitemoflist': this.replaceItemOfList,
+        'data_itemoflist': this.getItemOfList,
+        'data_lengthoflist': this.lengthOfList,
+        'data_listcontainsitem': this.listContainsItem
+    };
+};
+
+Scratch3DataBlocks.prototype.getVariable = function (args, util) {
+    var variable = util.target.lookupOrCreateVariable(args.VARIABLE);
+    return variable.value;
+};
+
+Scratch3DataBlocks.prototype.setVariableTo = function (args, util) {
+    var variable = util.target.lookupOrCreateVariable(args.VARIABLE);
+    variable.value = args.VALUE;
+};
+
+Scratch3DataBlocks.prototype.changeVariableBy = function (args, util) {
+    var variable = util.target.lookupOrCreateVariable(args.VARIABLE);
+    var castedValue = Cast.toNumber(variable.value);
+    var dValue = Cast.toNumber(args.VALUE);
+    variable.value = castedValue + dValue;
+};
+
+Scratch3DataBlocks.prototype.getList = function (args, util) {
```

Comment:
Ah, that makes sense. The 2.0 opcode is `contentsOfList:` so it might make sense to go with something involving "contents" for both the opcode and the function name.